### PR TITLE
Add framework for using external libs and add jslinq

### PR DESCRIPTION
**Pull request by sdmcraft**
This PR provides a framework for using external modules/libs (which are capable of getting executed in browser). It also includes a GH workflow which runs on PR submission and checks that module updates have been copied over to the code. For more details, please the the updates to README.md in this PR.

Additionally it adds `jslinq` module to support SQL like queries on JSON arrays. This is used for doing in-browser quesries on the index. More details on jslinq [here](https://github.com/maurobussini/jslinq/blob/master/README.md)

Test URLs:
- Original: https://www.sunstar.com
- Before: https://main--sunstar--hlxsites.hlx.live
- After: https://i-9b--sunstar--hlxsites.hlx.live

- [ ]  **If you are adding a new block or a variation to an existing block**, please fill below:

  Block library path: https://<branch>--sunstar--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/<your-block>&index=0

### CODE REVIEW

File: scripts/delayed.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 // eslint-disable-next-line import/no-cycle
 import { sampleRUM, isInternalPage } from './lib-franklin.js';
 // eslint-disable-next-line import/no-cycle
-import { getEnvType, loadConsentManager, loadScript } from './scripts.js';
+import {
```
Comment (bosschaert) [inline scripts/delayed.js:4]: I guess this file hasn't really changed so it doesn't need to be part of the PR?

**Review by bosschaert — APPROVED** [review]
Besides a handy framework, this PR also contains the jslinq library. I see the reason was given here: https://github.com/hlxsites/sunstar/pull/53#issuecomment-1732614010

Would it be possible to briefly outline how we apply jslinq?

**Comment by sdmcraft** [comment]
> Besides a handy framework, this PR also contains the jslinq library. I see the reason was given here: [#53 (comment)](https://github.com/hlxsites/sunstar/pull/53#issuecomment-1732614010)
> 
> Would it be possible to briefly outline how we apply jslinq?

You are right. I updated the PR title and description with details on `jslinq`. I have to use it for #9 so would add referenceable usage there.

**Comment by bosschaert** [comment]
> blabla

Just wondering where this comment came from?

**Comment by sdmcraft** [comment]
> > blabla
> 
> Just wondering where this comment came from?

That was me testing 🤣

**Comment by jindaliiita** [comment]
@sdmcraft it would be good if you can explain this PR in our sync up, because i am not properly able to understand this.

**Comment by sdmcraft** [comment]
> @sdmcraft it would be good if you can explain this PR in our sync up, because i am not properly able to understand this.

Have added some [details here](https://github.com/hlxsites/sunstar#adding-external-js-libraries). Also we can discuss in our sync meeting.

**Pull request merged**